src/Symfony/Component/HttpFoundation/ParameterBagUtils.php

```diff
@@ -0,0 +1,112 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\HttpFoundation;
+
+/**
+ * @internal
+ */
+final class ParameterBagUtils
```

Comment:
Is this really an internal class? Are we continuing to support looking "deep" by using this class, or are we totally deprecating that functionality and this class will also be deprecated and removed?